rasa_sdk/forms.py

```diff
@@ -689,3 +689,44 @@ def _get_entity_type_of_slot_to_fill(
                 return None
 
         return entity_type
+
+
+class FormSlotsValidatorAction(Action):
+    """An action that validates if every extracted slot is valid."""
+
+    def name(self) -> Text:
+        """Unique identifier of this action."""
+
+        raise NotImplementedError("An action must implement a name")
+
+    async def run(
+        self,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> List[EventType]:
+        """Execute the side effects of this action.
+
+        Args:
+            dispatcher: the dispatcher which is used to
+                send messages back to the user.
+            tracker: the state tracker for the current user.
+            domain: the bot's domain.
+        Returns:
+            A dictionary of ``rasa_sdk.events.Event`` instances.
+        """
+
+        slots_to_validate: Dict[Text, Any] = tracker.form_slots_to_validate()
+        validation_events = []
+
+        for slot_name, slot_value in slots_to_validate.items():
+            function_name = f"validate_{slot_name}"
+            fn = getattr(self, function_name)
+            if fn(tracker, domain, slot_value):
+                validation_events.append(SlotSet(slot_name, slot_value))
+            else:
+                # Return a `SlotSet` event with value `None` to indicate that this
```

Comment:
I think we should also print a warning for this. To be honest I'd just skip the validation for this slot then instead of marking it as invalid.